test/utils/structs/Merkletree.test.js

```diff
@@ -0,0 +1,61 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
+const { PANIC_CODES } = require('@nomicfoundation/hardhat-chai-matchers/panic');
+const { StandardMerkleTree } = require('@openzeppelin/merkle-tree');
+
+const makeTree = (leafs = [ethers.ZeroHash]) =>
+  StandardMerkleTree.of(
+    leafs.map(leaf => [leaf]),
+    ['bytes32'],
+    { sortLeaves: false },
+  );
+
+const DEPTH = 4n; // 16 slots
+const ZERO = makeTree().leafHash([ethers.ZeroHash]);
+
+async function fixture() {
+  return { mock: await ethers.deployContract('MerkleTreeMock', [DEPTH, ZERO]) };
+}
+
+describe('MerkleTree', function () {
+  beforeEach(async function () {
+    Object.assign(this, await loadFixture(fixture));
+  });
+
+  it('sets initial values at setup', async function () {
```

Comment:
We're missing a test case with a second setup after using the tree. We should have that test case since we're documenting it can be done.

Reply:
added